marshal/consumer.go

```diff
@@ -145,7 +145,15 @@ func (m *Marshaler) NewConsumer(topicNames []string, options ConsumerOptions) (*
 					}
 				}
 			}
+
+			// this check needs to be after iterating all partitions in a topic
+			if len(c.claimedTopics) >= options.MaximumClaims {
```

Comment:
Or just update the comment. "claimedTopics is only populated in topic claim mode" or something, so it's clear from inspection.